src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -59,23 +59,48 @@ pub async fn compact(
     payload: Vec<ImmutableMemtable>,
     compaction_catalog_manager_ref: CompactionCatalogManagerRef,
 ) -> HummockResult<UploadTaskOutput> {
-    let new_value_payload = payload.clone();
-    let new_value_future = async {
+    let table_ids_with_old_value: HashSet<TableId> = payload
+        .iter()
+        .filter(|imm| imm.has_old_value())
+        .map(|imm| imm.table_id)
+        .collect();
+    let mut non_log_store_new_value_payload = Vec::with_capacity(payload.len());
+    let mut log_store_new_value_payload = Vec::with_capacity(payload.len());
+    let mut old_value_payload = Vec::with_capacity(payload.len());
+    for imm in payload {
+        if table_ids_with_old_value.contains(&imm.table_id) {
+            log_store_new_value_payload.push(imm.clone());
+            if imm.has_old_value() {
```

Comment:
is `has_old_value` always true ?

Reply:
No. It's true when the table of the imm enables log store in the epoch.